packages/ember-data/tests/integration/setup-container-test.js

```diff
@@ -63,7 +67,7 @@ test("the deprecated adapter:_rest is resolved as adapter:rest", function() {
     deprecated = container.lookup('adapter:_rest');
   });
 
-  ok(deprecated === valid, "they should resolve to the same thing");
+  ok(deprecated._debugContainerKey === valid._debugContainerKey, "they should resolve to the same thing");
```

Comment:
The idea here is that both versions are the same actual class returned. That is why it used `===` before (and likely should still do).